common/lib/plugin_manager.ts

```diff
@@ -282,20 +283,26 @@ export class PluginManager {
   }
 
   getHostInfoByStrategy(role: HostRole, strategy: string, hosts?: HostInfo[]): HostInfo {
```

Comment:
Can you make the same optimization for acceptsStrategy since the code is very similar?